src/frontend/Pretty_printing.ml

```diff
@@ -75,18 +85,20 @@ let pp_comment ppf (lines, {Middle.Location_span.begin_loc= {col_num; _}; _}) =
   let trim_tail col_num lines =
     match lines with [] -> [] | hd :: tl -> hd :: trim (col_num - 2) tl
   in
-  Fmt.pf ppf "@[<v>/*%a*/@]" Fmt.(list string) (trim_tail col_num lines)
+  if is_block then
+    Fmt.pf ppf "@[<v>/*%a*/@]" Fmt.(list string) (trim_tail col_num lines)
+  else Fmt.pf ppf "@[<v>/*%a */@]" Fmt.(list string) lines
```

Comment:
Yes, this just reproduces a block comment at the moment.
A `//` comment needs a new line at the end.

Reply:
It seems like newlines are preserved in most cases, it is only a few edge cases where weird things happen, like in test/integration/good/ode_good.stan we have this section:

```stan
functions {
  real[] harm_osc_ode(real t,
                      real[] y,         // state
                      real[] theta,     // parameters
                      real[] x,         // data
                      int[] x_int) {    // integer data
```

which fails in the naive solution I suggested above.

Reply:
Exactly. That's where you need `Format.pp_force_newline`. But I'm afraid that knowing if you're in that context requires a large refactoring of the printer code.

Reply:
I've updated the tests which work for now, leaving out the exceptions so the tests will still fail in CI. This will require some thought, but I think we could try a special case if a comma and a line comment are on the same line?

Reply:
The difference is `pp_list_of` (inside parens) vs `pp_list_of_statements` (at toplevel). Those don't share as much code as I thought so maybe it's not so bad.

Reply:
Hopefully it is simpler then. Feel free to ping me if you want help/someone else to look at the problem